packages/headless/doc-parser/use-cases/product-recommendation.ts

```diff
@@ -0,0 +1,39 @@
+import {ActionLoaderConfiguration} from '../src/headless-export-resolvers/action-loader-resolver';
+import {ControllerConfiguration} from '../src/headless-export-resolvers/controller-resolver';
+import {EngineConfiguration} from '../src/headless-export-resolvers/engine-resolver';
+
+const controllers: ControllerConfiguration[] = [
+  {
+    initializer: 'buildContext',
+    samplePaths: {
+      react_fn: [
+        'packages/samples/headless-react/src/components/context/context.ts',
+      ],
+    },
+  },
+];
+
+const actionLoaders: ActionLoaderConfiguration[] = [
+  {
+    initializer: 'loadConfigurationActions',
+  },
+  {
+    initializer: 'loadProductRecommendationsActions',
+  },
+  {
+    initializer: 'loadContextActions',
+  },
+  {
+    initializer: 'loadSearchHubActions',
+  },
+];
```

Comment:
@francoislg, what event should clicks on product recommendations send to analytics? Do they reuse the search click action cause?

Reply:
My guess is that it should be the same as "normal" recommendation click event 🤔

Reply:
It would be like any click on anything search driven. For attribution, it needs to send the ID of the search that populated the list. As explained here https://docs.coveo.com/en/l41a1037/coveo-for-commerce/measure-events-on-a-listing-or-search-page#measure-product-clicks-on-a-search-driven-listing-page

Reply:
@SLangevin, I want to double check because search pages and regular recommendations are both search-api driven, but they send different click action causes to analytics.

The link you shared shows a search page, so a regular search click would be logical. However, in this case it is a product recommendation (e.g. cart recommendations, frequently viewed/bought etc.).

Do they all leverage the `documentOpen` action cause (like a search page), or the `recommendationOpen` action cause (like a recommendation), or do they have their own dedicated action cause(s)?

Reply:
I found [this table](https://docs.coveo.com/en/l5sb10500/coveo-for-commerce/event-parameters-and-associated-product-recommendation-models), but I'm not clear what the strings represent. For the cart recommender, are the strings `detail` and `purchase` meant to represent action causes for `custom` events?

For click events, it seems they are not required for most models, and only optional for two models? In the optional case, there is no action cause though,

Reply:
`documentOpen` and `recommendationOpen` are all event in the legacy CoveoUA Write. The new collect endpoint simply takes click and uses the search ID to attribute the click to the previous action.

The table you link are the required events for Product Recom to work.

Reply:
Ok, I am going to leave things as they are for now, and explore `collect`. Thanks Simon,